### PR TITLE
fix: latest react-native requires cli version >=4.x

### PR DESCRIPTION
**Pull request by tido64**
### Description

Nightlies have been failing because we're preventing the correct version of `@react-native-community/cli` and `metro` to be hoisted.

### Platforms affected

- [x] Android
- [x] iOS
- [x] macOS
- [x] Windows

### Test plan

```sh
yarn set-react-version master
cd example
yarn
yarn build:ios
```

### CODE REVIEW

File: scripts/set-react-version.js [hunk 1]
```diff
@@ -33,28 +33,19 @@ const profiles = {
     "react-native-windows": "0.63.14",
   },
   0.64: {
-    "@react-native-community/cli": "^5.0.1-alpha.0",
-    "@react-native-community/cli-platform-android": "^5.0.1-alpha.0",
-    "@react-native-community/cli-platform-ios": "^5.0.1-alpha.0",
```
Comment (kelset) [inline scripts/set-react-version.js:38]: I would argue that we should have >= 5 here...

File: scripts/set-react-version.js [hunk 2]
```diff
@@ -33,28 +33,19 @@ const profiles = {
...
     react: "17.0.1",
     "react-native": "0.64.0",
     "react-native-macos": undefined,
     "react-native-windows": "0.64.3",
   },
   master: {
-    "@react-native-community/cli": "^5.0.1-alpha.0",
-    "@react-native-community/cli-platform-android": "^5.0.1-alpha.0",
-    "@react-native-community/cli-platform-ios": "^5.0.1-alpha.0",
```
Comment (kelset) [inline scripts/set-react-version.js:47]: ...and >= 6 here...

File: scripts/set-react-version.js [hunk 3]
```diff
@@ -33,28 +33,19 @@ const profiles = {
...
-    react: "17.0.1",
+    react: "17.0.2",
     "react-native": "facebook/react-native#master",
     "react-native-macos": undefined,
     "react-native-windows": undefined,
   },
   nightly: {
-    "@react-native-community/cli": "^5.0.1-alpha.0",
-    "@react-native-community/cli-platform-android": "^5.0.1-alpha.0",
-    "@react-native-community/cli-platform-ios": "^5.0.1-alpha.0",
```
Comment (kelset) [inline scripts/set-react-version.js:56]: ...and >= 6 here.

**Review by kelset — APPROVED** [review]
my 2c$

File: scripts/set-react-version.js (hunk 1 above)
Comment (tido64) [reply scripts/set-react-version.js:38]: `react-native` pulls in the correct version if we just omit it. That's why I removed it here, and below.

**Review by kelset — APPROVED** [review]

**Pull request merged**